### PR TITLE
fix(metrics): Fixes bug that relies on metrics_query_args

### PR DESCRIPTION
**Pull request by ahmedetefy**
Prior to this PR, we were relying on creator of
a derived op to specifically define the args that will be required to be passed to the function generating the snql or the post query function that applies post query processing logic, which is error prone. In this PR, we instead fetch those function arguments directly and fetch their equivalent in the passed params from `MetricField` dynamically

### CODE REVIEW

File: src/sentry/snuba/metrics/fields/base.py [hunk 1]
```diff
@@ -359,10 +359,23 @@ def run_post_query_function(
         else:
             subdata = data[alias][idx]
 
-        compute_func_dict = {"data": subdata}
-        if self.metrics_query_args is not None and params is not None:
-            for field in self.metrics_query_args:
-                compute_func_dict[field] = params.get(field)
+        # Fetch the function args
+        metrics_query_args = set(inspect.signature(self.post_query_func).parameters.keys())
```
Comment (andriisoldatenko) [inline src/sentry/snuba/metrics/fields/base.py:363]:
```suggestion
        metrics_query_args = inspect.signature(self.post_query_func).parameters.keys()
```
I think set is redundant here:
```
In [3]: def foo(a, b): pass

In [4]: inspect.signature(foo)
Out[4]: <Signature (a, b)>

In [5]: inspect.signature(foo).parameters
Out[5]: mappingproxy({'a': <Parameter "a">, 'b': <Parameter "b">})

In [6]: inspect.signature(foo).parameters.keys()
Out[6]: odict_keys(['a', 'b'])

In [7]: 1 in inspect.signature(foo).parameters.keys()
Out[7]: False
```

File: src/sentry/snuba/metrics/fields/base.py [hunk 2]
```diff
@@ -375,11 +388,17 @@ def run_post_query_function(
     def generate_filter_snql_conditions(
         self, org_id: int, use_case_id: UseCaseKey, params: Optional[MetricOperationParams] = None
     ) -> Optional[Function]:
-        kwargs = {"org_id": org_id}
-        if self.metrics_query_args is not None and params is not None:
-            for field in self.metrics_query_args:
-                kwargs[field] = params.get(field)  # type: ignore
-
+        # Fetch the function args
+        metrics_query_args = set(inspect.signature(self.filter_conditions_func).parameters.keys())
```
Comment (andriisoldatenko) [inline src/sentry/snuba/metrics/fields/base.py:392]:
```suggestion
        metrics_query_args = inspect.signature(self.filter_conditions_func).parameters.keys()
```

Comment (andriisoldatenko) [reply src/sentry/snuba/metrics/fields/base.py:392]: same here.

Comment (ahmedetefy) [reply src/sentry/snuba/metrics/fields/base.py]: fixed

File: src/sentry/snuba/metrics/fields/base.py (hunk 1 above)
Comment (ahmedetefy) [reply src/sentry/snuba/metrics/fields/base.py]: fixed

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**